packages/core/src/hooks/useContractRead.ts

```diff
@@ -75,6 +78,16 @@ export function useContractRead({
     [enabled_, contract, functionName, args],
   );
 
+  const refetchInterval = useMemo(
```

Comment:
useMemo is unnecessary here, it will probably be slower
also i'd think this is more readable

`refetchInterval_ ?? (blockIdentifier === BlockTag.pending && watch ? DEFAULT_FETCH_INTERVAL : undefined)`